llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -31,6 +31,11 @@ using namespace llvm;
 #define DEBUG_TYPE "aarch64-isel"
 #define PASS_NAME "AArch64 Instruction Selection"
 
+// https://github.com/llvm/llvm-project/issues/114425
+#if defined(_WIN32) && !defined(__clang__) && !defined(NDEBUG)
```

Comment:
Should this test for MSVC specifically, rather than `!defined(__clang__)` ?

Reply:
I've matched #110986 - which IIRC is doing this to avoid it firing on clang-cl

Reply:
Yes - clang-cl defines the _MSC_VER macro as well as __clang__ :https://clang.godbolt.org/z/fo9aMYosK

Reply:
But this condition as it stands, `defined(_WIN32) && !defined(__clang__)` will also match for GCC.

The correct one for singling out MSVC but not clang in msvc mode, would be `defined(_MSC_VER) && !defined(__clang__)`.

Reply:
I'll create a patch